csharp/extractor/Semmle.Extraction.CSharp/Entities/CommentBlock.cs

```diff
@@ -11,11 +11,12 @@ private CommentBlock(Context cx, Comments.CommentBlock init)
         public override void Populate(TextWriter trapFile)
         {
             trapFile.commentblock(this);
-            if (!Context.OnlyScaffold)
+            Symbol.CommentLines.ForEach((l, child) => trapFile.commentblock_child(this, l, child));
+            if (Context.OnlyScaffold)
```

Comment:
I would have thought that we could simply move this early return up to the beginning of `Populate`?

Reply:
I would like to move this to the begging on the populate method and then also guard the `BindTo`  method. This should be internally consistent and not cause any dangling references (I will also push the scaffolding guard from CachedEntitiy to the Context for comment binding).
Will add one more commit for this.

Also note that for TypeMentions, the guard has been moved to the top of the populate method (as creating these will not cause any dangling references). Most places in the populate methods the TypeMention creation has been "moved below" the scaffolding early return check (and thus not be created), but I haven't done so in all places as this will create a large amount of extra guarding.

Reply:
Hmm, I will also put in the guards the places where you suggest to avoid all the string processing before the actual comments are created.